### PR TITLE
Fix incremental compilation in javalib

### LINKED ISSUE #599: Disable incremental compilation for javalib
see https://github.com/sbt/sbt/issues/3065

I tried what they have in Scala.js but this gave me null pointer exception in NirCodeGen

https://github.com/scala-native/scala-native/blob/b211e202c22846c64e9ea2222c0b8062d85b73e8/nscplugin/src/main/scala/scala/scalanative/nscplugin/NirCodeGen.scala#L1030

### PR DESCRIPTION
**Pull request by Duhemm**
Fixes #599

### CODE REVIEW

File: build.sbt [hunk 1]
```diff
@@ -299,7 +299,15 @@ lazy val javalib =
     .settings(libSettings)
     .settings(mavenPublishSettings)
     .settings(
-      sources in doc in Compile := Nil // doc generation currently broken
+      sources in doc in Compile := Nil, // doc generation currently broken
+      scalacOptions in Compile := {
```
Comment (MasseGuillaume) [inline build.sbt]: maybe add a comment. why we need this and how does it works.

Comment (Duhemm) [inline build.sbt]: Done.

**Comment by MasseGuillaume** [comment]
@Duhemm could this be useful for Scala.js ?

**Comment by sjrd** [comment]
It could be *useful*, but I'm too wary of messing with the bootclasspath to accept such a solution in our build ;)

**Comment by densh** [comment]
@sjrd We'll beta-test the approach, and see how it goes. :)

**Pull request merged**